src/Plugin/CKEditorPlugin/DrupalEntity.php

```diff
@@ -42,13 +51,27 @@ class DrupalEntity extends CKEditorPluginBase {
    *   The plugin_id for the plugin instance.
    * @param mixed $plugin_definition
    *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\Query\QueryFactory $entity_query
+   *   The entity query.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, QueryFactory $entity_query) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
 
-    $this->embedButtons = \Drupal::entityQuery('embed_button')->execute();
+    $this->entityQuery = $entity_query;
+    $this->embedButtons = $this->entityQuery->get('embed_button')->execute();
```

Comment:
There's no need to store entity_query object. You can just do `$this->embedButtons = entity_query->get()`.